src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -423,6 +421,97 @@ public CompletableFuture<Either<List<? extends Location>, List<? extends Locatio
+                if (target.isPresent()) {
+                    shapeToSerialize = target.get();
+                } else {
+                    shapeToSerialize = initialShape;
+                }
+            } else {
+                Optional<Shape> shapeMatchedByNameOnly = model.shapes()
+                        .filter(shape -> !shape.isMemberShape())
+                        .filter(shape -> shape.getId().getName().equals(token))
+                        .findAny();
+                if (shapeMatchedByNameOnly.isPresent()) {
+                    shapeToSerialize = shapeMatchedByNameOnly.get();
+                }
+            }
+        } catch (Exception e) {
+            LspLog.println("Failed to determine hover content: " + e);
+        }
+
+        // If a shape to serialize has been found, serialize the shape and set it to the hover contents.
+        if (shapeToSerialize != null) {
+            content.setValue("```smithy\n" + getSerializedShape(shapeToSerialize, model) + "\n```");
+        }
+        hover.setContents(content);
+        return Utils.completableFuture(hover);
+    }
+
+    // Finds the first non-member neighbor shape or trait applied to a member whose name matches the token.
+    private Optional<Shape> getTargetShape(Shape initialShape, String token, Model model) {
+        LspLog.println("Finding target of: " + initialShape);
+        Walker shapeWalker = new Walker(NeighborProviderIndex.of(model).getProvider());
+        return shapeWalker.walkShapes(initialShape).stream()
+                .flatMap(shape -> {
+                    if (shape.isMemberShape()) {
+                        return shape.getAllTraits().values().stream()
+                                .map(trait -> trait.toShapeId());
+                    } else {
+                        return Stream.of(shape.getId());
+                    }
+                })
+                .filter(shapeId -> shapeId.getName().equals(token))
+                .map(shapeId -> model.getShape(shapeId).get())
+                .findFirst();
+    }
+
+    private String getSerializedShape(Shape shape, Model model) {
+        SmithyIdlModelSerializer serializer = SmithyIdlModelSerializer.builder()
+                .metadataFilter(key -> false)
+                .shapeFilter(s -> s.getId().equals(shape.getId()))
+                .serializePrelude().build();
+        Map<Path, String> serialized = serializer.serialize(model);
+        Path path = Paths.get(shape.getId().getNamespace() + ".smithy");
+        return stripPreamble(serialized.get(path));
+    }
+
+
+    private String stripPreamble(String serializedModel) {
+        List<String> lines = Arrays.asList(serializedModel.split("\n"));
+        int preambleEnd = lines.size();
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            if (line.startsWith("$version: ") || line.startsWith("namespace ") || line.startsWith("use ")
+                    || line.isEmpty()) {
```

Comment:
From my understanding, the [model serializer](https://github.com/awslabs/smithy/blob/7d421894268ed68cb6ae5bf536a2a7850ad06784/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java) used to generate the rendered markdown, used [here](https://github.com/awslabs/smithy-language-server/pull/63/files/9f00da2ef6843ac9960c0a635c2fd7bab9036332#diff-e324d5c1b7824ab09a840e49300b646d6cdc7bf69a4eb19666ff08e8adc70062R489) before the `stripPreamble` method, uses the _shape_, not what is present in the `.smithy` file. It seems to always generate `$version: "2.0"` and the namespace of the shape, along with any `use` statements that the shape depends on. At the moment, I don't think there is a way for the LSP to provide a shape to the serializer that depends on other control statements and thus the serializer will not produce any other control statements in the rendered markdown, but I think it is worthwhile to handle these cases anyways for potential changes down the road.

Two other things this brings up: 
1. Is the name `stripPreamble` misleading? The method is really stripping the control and metadata sections, along with the `namespace` and `use` statements. 
2. Going off of 1, do we actually want to get rid of the `use` and `namespace` statements? These provide useful information about the shape being hovered over. At the moment, since `$version` is always "2.0", I'm not sure how helpful that is, in particular because any 1.0 models are serialized as 2.0 models, so I think we should keep that out of the rendered markdown, but I think we should reconsider keeping `use` and `namespace` statements in the rendered markdown.

Reply:
Although, having `$version: "2.0"` in the rendered markdown could actually be helpful because it would give an indication as to why there is different syntax in the rendered markdown if the source file is 1.0 syntax.

Reply:
From offline conversation, it looks like there are a few aspects:

- Stripping the preamble removes those lines from the hover content, which filters out info, both irrelevant or important
- If the preamble is not stripped, current tests will break due to position changes

Personally, I think it is okay to keep the preamble, but we should test more extreme cases and see how it affects the hover content (numerous amounts of control statements, use statements, empty lines, etc.).

Alternatively, is there a way to strip unwanted lines based on the input serialized shape? That way, we don't have to worry about what is considered the preamble.

It looks like the [`SmithyIdlModelSerializer.serializeHeader()` method](https://github.com/syall/smithy/blob/7d421894268ed68cb6ae5bf536a2a7850ad06784/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java#L197-L228) is called and is equivalent to the preamble we want to strip. We could try to find the last line written by the this method or the first line written by [`codeWriter.toString()`](https://github.com/syall/smithy/blob/7d421894268ed68cb6ae5bf536a2a7850ad06784/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java#L169) (unsure of written shape order though) that follows it in [`SmithyIdlModelSerializer.serialize()`](https://github.com/syall/smithy/blob/7d421894268ed68cb6ae5bf536a2a7850ad06784/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java#L147-L170).

If we want to be precise about stripping the preamble, it might be better to create a method that is identical to `SmithyIdlModelSerializer.serialize()` but does not serialize the preamble (omits `SmithyIdlModelSerializer.serializeHeader()`.

Reply:
I think that all info included in the preamble is relevant. The `$version` indicates that the rendered markdown is using IDL 2.0 syntax, the `namespace` indicates where the hovered shape comes from, and the `use` statements show what the hovered shape depends on.

With regards to the [`SmithyIdlModelSerializer.serializeHeader()` method](https://github.com/syall/smithy/blob/7d421894268ed68cb6ae5bf536a2a7850ad06784/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java#L197-L228), metadata is filtered out (see [here](https://github.com/syall/smithy/blob/7d421894268ed68cb6ae5bf536a2a7850ad06784/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java#L206)) using the `metadataFilter` property, which is being set to filter out all metadata [when the serializer is built](https://github.com/awslabs/smithy-language-server/pull/63/files#diff-e324d5c1b7824ab09a840e49300b646d6cdc7bf69a4eb19666ff08e8adc70062R482).

Do we have a good reason for stripping the preamble in the first place? It comes down to whether we want to attempt to replicate the hovered shape in the rendered markdown exactly as it is in the source file (which is technically not possible because there can be external dependencies for which we don't have the source file), or are we trying to provide all the information potentially useful to understand the hovered shape. For example, say we have two different Smithy files:
```
$version: "2.0"

namespace com.foo

use com.other#OtherStructure

structure SomeStructure {
    someMember: OtherStructure
}
```
```
$version: "2.0"

namespace com.other

use com.external#ExternalStructure

structure OtherStructure {
    someOtherMember: ExternalStructure
}
```
When hovering over the `someMember` target `OtherStructure` here is the rendered markdown both with and without the preamble:
No preamble:
```
structure OtherStructure {
    someOtherMember: ExternalStructure
}
```
With preamble:
```
$version: "2.0"

namespace com.other

use com.external#ExternalStructure

structure OtherStructure {
    someOtherMember: ExternalStructure
}
```
With the preamble, you get way more information, but it depends on if thats what we want to provide.

We can add more test cases to verify that there isn't any unexpected content in the preamble, but it should be noted that the generated markdown is only dependent on what is stored within the model (in addition to the version number). From the [IDL docs](https://awslabs.github.io/smithy/2.0/spec/idl.html#control-section): _"Because control statements influence parsing, they MUST appear at the beginning of a file before any other statements and have no effect on the [semantic model](https://awslabs.github.io/smithy/2.0/spec/model.html#semantic-model)"_ and _"Implementations MUST ignore unknown control statements."_. From my understanding, empty lines are not stored in the model, and any `use` statements will only be accessible within the serializer if the given shape depends on them. We should take these factors into consideration when designing the tests, verifying that the correct `use` and `namespace` statements are present, but I don't believe that any extraneous content within source files' preambles will have an impact on the rendered markdown. I will add some test cases to verify that shortly.

Reply:
0b9261979200b1ad2214036d7c8dab93859457f7 should fix. Leaving the conversation open for now to make sure.